discretize/base/base_mesh.py

```diff
@@ -1880,6 +1883,415 @@ def get_edge_inner_product(
+            \approx \sum_i \int_{\ell_i} \, \vec{u}_\parallel \cdot \lambda \vec{v}_\parallel \, ds
+            = \mathbf{u^T \, M_\lambda \, w}
+
+        where the diagnostic properties on mesh edges (i.e. the model) are stored within an array of the form:
+
+        .. math::
+            \boldsymbol{\lambda} = \begin{bmatrix} \boldsymbol{\lambda_x} \\
+            \boldsymbol{\lambda_y} \\ \boldsymbol{\lambda_z} \end{bmatrix}
+
+        Examples
+        --------
+        Here we provide an example of edge inner product line matrix.
+        For simplicity, we will work on a 2 x 2 x 2 tensor mesh.
+        As seen below, we begin by constructing and imaging the basic
+        edge inner product line matrix.
+
+        >>> from discretize import TensorMesh
+        >>> import matplotlib.pyplot as plt
+        >>> import numpy as np
+        >>> import matplotlib as mpl
+
+        >>> h = np.ones(2)
+        >>> mesh = TensorMesh([h, h, h])
+        >>> Me = mesh.get_edge_inner_product_line()
+
+        >>> fig = plt.figure(figsize=(6, 6))
+        >>> ax = fig.add_subplot(111)
+        >>> ax.imshow(Me.todense())
+        >>> ax.set_title('Basic Edge Inner Product Line Matrix', fontsize=18)
+        >>> plt.show()
+
+        Next, we consider the case where the physical properties
+        are defined by diagnostic properties on mesh edges. For the isotropic case,
+        we show the physical property tensor for a single cell.
+
+        Define the diagnostic property values for x, y and z faces.
+
+        >>> tau_x, tau_y, tau_z = 3, 2, 1
+
+        Here construct and image the edge inner product line matrix for the isotropic case.
+        Spy plots are used to demonstrate the sparsity of the matrix.
+
+        >>> tau = np.r_[tau_x * mesh.n_edges_x, tau_y * mesh.n_edges_y, tau_z * mesh.n_edges_z]
+        >>> M = mesh.get_edge_inner_product_line(tau)
+
+        Then plot the sparse representation,
+
+        >>> fig = plt.figure(figsize=(4, 4))
+        >>> ax1 = fig.add_subplot(111)
+        >>> ax1.spy(M, ms=5)
+        >>> ax1.set_title("M (isotropic)", fontsize=16)
+        >>> plt.show()
+        """
+        try:
+            if model is None:
+                model = np.ones(self.nE)
+
+            if invert_model:
+                model = 1.0 / model
+
+            if is_scalar(model):
+                model = model * np.ones(self.nE)
+
+            # Isotropic case only
+            if model.size == self.nE:
+                Lprop = self.edge_lengths * mkvc(model)
+                M = sdiag(Lprop)
+
+            else:
+                raise Exception(
+                    "Unexpected shape of tensor: {}".format(model.shape),
+                    "Must be scalar or have length equal to total number of edges.",
+                )
+
+        except NotImplementedError:
+            raise NotImplementedError(
+                f"get_edge_inner_product_line not implemented for {type(self)}"
+            )
```

Comment:
I would replace these lines with basically the same thing as I put for the face innerproduct surface function.